### DISCUSSION TITLE
Documentation clarification for running tasks on one server

### DISCUSSION THREAD

**Discussion by aanderse** [laravel/framework#33216] [General] [answered]
Is the documentation about requiring `memcached` or `redis` as a cache driver correct (as in up to date) if you want to [run tasks on one server](https://laravel.com/docs/7.x/scheduling#running-tasks-on-one-server)?

I see that `laravel` more [recently](https://github.com/laravel/framework/pull/32639) added [atomic locks](https://laravel.com/docs/7.x/cache#atomic-locks) to the `database` driver. This makes me wonder if the `database` cache driver actually would work and the documentation may not have been updated.

**Comment by aanderse** [comment, marked as answer]
ping @themsaid based on an IRC conversation.

**Reply by themsaid** [reply]
I don't get your question. Can you elaborate?

**Reply by aanderse** [reply]
Sure. I would like to know if after https://github.com/laravel/framework/pull/32639 the following statement (https://laravel.com/docs/7.x/scheduling#running-tasks-on-one-server) is still correct:
> To utilize this feature, your application must be using the memcached or redis cache driver as your application's default cache driver. In addition, all servers must be communicating with the same central cache server.

My understanding is that running tasks on one server uses atomic locks from the cache to work. Previously the `database` backend for cache did not support atomic locks so the above statement makes sense to me. Since the `database` backend for caches now supports atomic locks I would reason that if I'm using the `database` backend for my cache I can run tasks on a single server and that the documentation is out of date.

For additional context I'm talking about the scenario where I have a `laravel` site running on multiple servers and each of these servers is running a scheduler. I want to be able to have the scheduler avoid running the same job on all my servers using the `onOneServer` method.

Does this clarify the question sufficiently?

**Reply by aanderse** [reply]
@themsaid I'll use a scenario which will hopefully further clarify:

I am running my `laravel` web app on both `server A` and `server B`. There is a load balancer which round robin distributes requests to my site between `server A` and `server B`. I have a scheduled job in my application, but I only want the job to run [on one server](https://laravel.com/docs/7.x/scheduling#running-tasks-on-one-server). According to `laravel` documentation I have to use the `memcached` or `redis` cache driver:
> To utilize this feature, your application must be using the memcached or redis cache driver as your application's default cache driver. In addition, all servers must be communicating with the same central cache server.

I want to confirm if this statement is entirely true. My assumption is that running a `laravel` scheduled task on one server relies on atomic locks, provided by the cache driver. Since the `database` cache driver [now supports atomic locks](https://github.com/laravel/framework/pull/32639) is the documentation out of date and the `database` cache driver would be sufficient to use the `onOneServer` method on scheduled jobs?

Maybe @taylorotwell has some insight as they added `database` cache driver support for atomic locks.

**Reply by aanderse** [reply]
Seems like @taylorotwell implicitly confirmed this in https://github.com/laravel/docs/pull/6138 :+1:
